core/src/test/scala/integration/kafka/api/BaseAdminIntegrationTest.scala

```diff
@@ -189,6 +192,15 @@ abstract class BaseAdminIntegrationTest extends IntegrationTestHarness with Logg
 
   override def modifyConfigs(configs: Seq[Properties]): Unit = {
     super.modifyConfigs(configs)
+    // For testCreateTopicsReturnsConfigs, set some static broker configurations so that we can
+    // verify that they show up in the "configs" output of CreateTopics.
```

Comment:
Yes, this is kind of a problem with the old test harness. I agree that the non-locality could be confusing here. However, I think it would be a lot of repeated boilerplate code to create a new test file just for this one test case.

As a compromise, I added some JavaDoc to `testCreateTopicsReturnsConfigs` so that anyone modifying that test can see where the configurations are being set....